src/main/java/com/capgemini/fs/coindashboard/apiCommunicator/coinMarketCapCommunicator/CoinMarketCapResponseParser.java

```diff
@@ -0,0 +1,167 @@
+      {
+        coin = new SimpleEntry<>(coin.getKey(), coin.getValue().get(0));
+      }
+      String name = coin.getKey();
+      String symbol = coin.getValue().get("symbol").asText();
+
+      Map<String, QuoteDto> quoteDtos = new LinkedHashMap<>();
+      Map<String, ObjectNode> quotesConverted = mapper.convertValue(coin.getValue().get("quote"),
+          new TypeReference<>() {
+          });
+
+      for (Map.Entry<String, ObjectNode> quote : quotesConverted.entrySet()) {
+        var currentPrice = quote.getValue().get("price").asDouble();
+        var timestamp = TimeFormatter.convertStringToTimestamp(
+                quote.getValue().get("last_updated").asText(), "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+            .getTime();
+        var pctChange1h = quote.getValue().get("percent_change_1h").asDouble();
+        var pctChange24h = quote.getValue().get("percent_change_24h").asDouble();
+        var pctChange7d = quote.getValue().get("percent_change_7d").asDouble();
+        var pctChange30d = quote.getValue().get("percent_change_30d").asDouble();
+        var nominalChange1h = this.calculateNominalDelta(currentPrice, pctChange1h);
+        var nominalChange24h = this.calculateNominalDelta(currentPrice, pctChange24h);
+        var nominalChange7d = this.calculateNominalDelta(currentPrice, pctChange7d);
+        var nominalChange30d = this.calculateNominalDelta(currentPrice, pctChange30d);
+        List<DeltaDto> deltas = new ArrayList<>() {{
+          add(new DeltaDto(IntervalEnum.ONE_HOUR, pctChange1h, nominalChange1h));
+          add(new DeltaDto(IntervalEnum.ONE_DAY, pctChange24h, nominalChange24h));
+          add(new DeltaDto(IntervalEnum.SEVEN_DAY, pctChange7d, nominalChange7d));
+          add(new DeltaDto(IntervalEnum.THIRTY_DAY, pctChange30d, nominalChange30d));
+        }};
+
+        quoteDtos.put(quote.getKey(), new QuoteDto(new ArrayList<>() {{
+          add(new PriceDto(currentPrice, timestamp));
+        }}, deltas, timestamp));
+      }
+      result.add(new CoinMarketDataDto(
+          name, symbol, quoteDtos));
+    }
+    return result;
+  }
+
+
+  public CoinMarketDataResult parseQuoteHistoricalResult(JsonNode apiResponseBody) {
+    CoinMarketDataResult result = new CoinMarketDataResult();
+    try {
+      String error = this.parseStatus(apiResponseBody.get("status"));
+      if (error != null) {
+        result.setStatus(ResultStatus.FAILURE);
+        result.setErrorMessage(error);
+        return result;
+      }
+
+      result.setStatus(ResultStatus.SUCCESS);
+      result.setCoinMarketDataDTOS(
+          this.parseCoinsQuoteHistoricalResult(apiResponseBody.get("data")));
+      return result;
+    } catch (Exception e) {
+      log.error(e.getMessage());
+      result.setStatus(ResultStatus.FAILURE);
+      result.setErrorMessage(e.getMessage());
+      return result;
+    }
+  }
+
+  public ArrayList<CoinMarketDataDto> parseCoinsQuoteHistoricalResult(JsonNode data)
+      throws ParseException {
```

Comment:
This method consists too much different responsibilities. In quick refactor I have extracted a bunch of methods that made hierarchy of actions much more visible:
- `extractCoinMarketDataDtos`
- `extractCoinMarketDataDto`
- `extractQuoteDtos`
- `extractQuoteDto`
- `convertNodeToMap`
I don't mention it as preferable division of code, I just did it to make it more readable for me (so I don't share the result). Please consider splitting this code into multiple parsers that create respective objects. Or some other solution to make this code more readable.